### PR TITLE
Chore: Replace 'walkes' with 'estree-walker'

### LINKED ISSUE #1210: Remove dependency on "walkes"
Starting in version 8 of the package it looks like "walkes" was added as a project dependency, although I cannot find out where it is used. I am not sure if this is needed?

That library is LGPL-3 licensed and can make this project unusable in certain licensing situations starting in version 8.

### PR DESCRIPTION
**Pull request by eragon512**
@sapegin could you please look at this PR and check if this is acceptable?

I'm concerned that this might break existing functionality, as the AST walking is a core part of the code, and the implemented code technically breaks the tests, however, this seems to be a popular fix, so opening this up for further discussion

Fixes #1210

### CODE REVIEW

**Comment by semi-sentient** [comment]
It looks like "walkes" is still listed as a dev dependency. Shouldn't that be removed as well?

**Comment by eragon512** [comment]
@semi-sentient yeah, you're right, I'll do that.
Is this PR under consideration? because I had closed it off, due to the fact that estree-walker was a pretty big module.
Is the license issue important enough to increase the bundle size? :o

**Comment by semi-sentient** [comment]
I was just sort of snooping around and noticed that walkes was still listed as a dependency. Since it has a more restrictive license we can't import it for use here @Amazon so just wanted to point that out as I'd very much love to use react-styleguidist on my current project. ;)

**Comment by richardwestenra** [comment]
@eragon512 What, in your view, is necessary to get this PR ready for consideration? This dependency license has been an issue for my team too.

**Comment by ottis** [comment]
Keen to get this resolved too, similar situation as @semi-sentient i.e. we'd love to use this but the license of walkes is restrictive 😢

**Comment by eragon512** [comment]
@richardwestenra @ottis I didn't realise this was such a major issue for you guys 😅 . I'll try to take this up again, seeing the demand for it

My main concern is that the replacement in this PR (`estree-walker`) *might* break existing functionality, but I guess @sapegin would be a better person to weigh in here, as I'm not very familiar with ASTs and such

**Comment by eragon512** [comment]
Re-opening for active consideration and further work

File: src/client/utils/__tests__/compileCode.spec.js [hunk 1]
```diff
@@ -14,10 +14,7 @@ var bar = baz.bar;"
 
 	test('transform imports to require()', () => {
 		const result = compileCode(`import foo from 'bar'`, compilerConfig);
-		expect(result).toMatchInlineSnapshot(`
-"const bar$0 = require('bar');
-const foo = bar$0.default || bar$0;"
-`);
+		expect(result).toMatchInlineSnapshot(`"import foo from 'bar'"`);
```
Comment (sapegin) [inline src/client/utils/__tests__/compileCode.spec.js]: These changes are very suspicious, it means the compilation doesn't happen: the resulting code is the same as the source ;-£

File: src/client/utils/__tests__/transpileImports.spec.js [hunk 2]
```diff
@@ -42,17 +32,13 @@ import E from 'snake'
 "/**
 * Some important comment
 */
-require('dog');
+import 'dog'
```
Comment (sapegin) [inline src/client/utils/__tests__/transpileImports.spec.js]: Also regular `require()` should still work.

File: src/client/utils/transpileImports.js [hunk 3]
```diff
@@ -21,7 +21,7 @@ export default function transpileImports(code: string): string {
 	}
 
 	let offset = 0;
-	walkes(ast, {
+	walk(ast, {
```
Comment (sapegin) [inline src/client/utils/transpileImports.js]: Looks like the API is different and it should be something like this:

```js
walk( ast, {
			enter ( node ) {
				if (node.type === 'ImportDeclaration') {
                                     // do the work
			        }
			},
})


```

File: src/client/utils/__tests__/compileCode.spec.js (hunk 1 above)
Comment (eragon512) [reply src/client/utils/__tests__/compileCode.spec.js]: My fears exactly

**Comment by eragon512** [comment]
@sapegin please check the code when you get time, `estree-walker` seems to be a perfect drop-in replacement

The earlier suspicious test snapshots were my bad, I hadn't completely replaced the `walkes` dependency before 😅

File: src/client/utils/transpileImports.js (hunk 3 above)
Comment (eragon512) [reply src/client/utils/transpileImports.js]: Implemented 👍

**Comment by sapegin** [comment]
That look good! 

Could you please replace Flow types here?

https://github.com/styleguidist/react-styleguidist/tree/master/flow-typed

**Comment by eragon512** [comment]
> That look good!
> 
> Could you please replace Flow types here?
> 
> https://github.com/styleguidist/react-styleguidist/tree/master/flow-typed

Replaced the flow types, as requested

**Comment by sapegin** [comment]
Awesome!

If you merge these two conditions, it should fix the code coverage check:

```
 if (node.type !== 'ImportDeclaration') {
   return;
 }
 if (!node.source) {
  return;
 }
```

**Comment by eragon512** [comment]
done, please confirm

**Comment by sapegin** [comment]
Thanks, merging!

**Comment by eragon512** [comment]
@semi-sentient @richardwestenra @ottis here you go!

**Comment by Swatinem** [comment]
Hm, just out of boredom I was checking all my old npm packages and was like *whaaaaaaa* when I saw this package getting ~50k downloads a week while its been basically unmaintained since 4-6 years.
… and then I landed here and this made my day :-D

**Pull request merged**